automate-collections-updates/automate_collections_updates.py

```diff
@@ -0,0 +1,154 @@
+                        open(os.path.join(coll_dir, "changelogs/changelog.yaml"))
+                    )
+                    item = re.search(
+                        value, yaml.dump(changelog["releases"][coll_latest_ver])
+                    )
+                    if item is not None:
+                        print(
+                            f"{item.group()} is mentioned in latest {coll_name} changelog."
+                        )
+                        """Add coll_name key to collection_tarballs"""
+                        collection_tarballs[coll_name] = ""
+            if coll_name in collection_tarballs.keys():
+                tar = tarfile.open(
+                    coll_name.replace(".", "-") + "-" + coll_latest_ver + ".tar.gz",
+                    "w:gz",
+                )
+                """Create collection tarballs"""
+                for name in os.listdir(coll_dir):
+                    tar.add(name=str(os.path.join(coll_dir, name)), arcname=str(name))
+                tar.close()
+                collection_tarballs[coll_name] = os.path.basename(tar.name)
+                print(f"Collection tarball {os.path.basename(tar.name)} is created.")
+            else:
+                print(
+                    f"Latest {coll_name} changelog does not list updates for the vendored modules."
+                )
+        finally:
+            shutil.rmtree(coll_install_dir)
+    if len(collection_tarballs) != 0:
+        try:
+            subprocess.run(
+                [
+                    "git",
+                    "clone",
+                    repo_url,
+                    "--quiet",
+                    "--branch",
+                    branch,
+                    "--single-branch",
+                ]
+            )
+            """Move tarballs to the repo directory"""
+            for tarball in collection_tarballs.values():
+                shutil.move(os.path.abspath(tarball), os.path.join(repo, tarball))
+            """Upload new sources"""
+            os.chdir(repo)
+            subprocess.run(
+                [pkg_command, "upload", " ".join(collection_tarballs.values())]
+            )
+            """Replace sources in the spec file"""
+            for collection, tarball in collection_tarballs.items():
+                with open("linux-system-roles.spec", "r") as f:
+                    content = f.read()
+                    re.sub(collection + ".*$", tarball, content, flags=re.M)
+            out = subprocess.run(
+                [pkg_command, "scratch-build", "--srpm", "--nowait"],
+                stdout=subprocess.PIPE,
+                encoding="utf-8",
+            )
+            pat = "Task info: (.*$)"
+            build_url = re.search(pat, out.stdout).group(1)
+            print(f"Build info: {build_url}")
+            # TODO: Share build_url somewhere
+        finally:
+            print("Cleaning up the downloaded repository")
+            os.chdir("..")
```

Comment:
which directory is this script supposed to be run from?  I think you'll have to be careful about paths if you assume a certain directory, or don't assume a cwd anywhere - all of the `open` calls should construct the full path and filename - use the `cwd` argument of `subprocess.run` to run the command in a certain directory.

Reply:
Good point, I'll fix this